services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/ResponseBatchConfiguration.java

```diff
@@ -50,76 +46,60 @@ public final class ResponseBatchConfiguration {
     private final Integer maxBatchItems;
     private final Integer maxInflightReceiveBatches;
     private final Integer maxDoneReceiveBatches;
 
-    public ResponseBatchConfiguration(BatchOverrideConfiguration overrideConfiguration) {
-        this.visibilityTimeout = overrideConfiguration != null && overrideConfiguration.visibilityTimeout() != null
-                                 ? overrideConfiguration.visibilityTimeout()
+    private ResponseBatchConfiguration(Builder builder) {
+        this.visibilityTimeout = builder.visibilityTimeout != null
+                                 ? builder.visibilityTimeout
                                  : VISIBILITY_TIMEOUT_SECONDS_DEFAULT;
 
-        this.longPollWaitTimeout = overrideConfiguration != null && overrideConfiguration.longPollWaitTimeout() != null
-                                   ? overrideConfiguration.longPollWaitTimeout()
-                                   : LONG_POLL_WAIT_TIMEOUT_DEFAULT;
-
-        this.minReceiveWaitTime = overrideConfiguration != null && overrideConfiguration.minReceiveWaitTime() != null
-                                  ? overrideConfiguration.minReceiveWaitTime()
+        this.messageMinWaitDuration = builder.messageMinWaitDuration != null
+                                  ? builder.messageMinWaitDuration
                                   : MIN_RECEIVE_WAIT_TIME_MS_DEFAULT;
 
-        this.messageSystemAttributeValues = overrideConfiguration != null
-                                            && overrideConfiguration.messageSystemAttributeName() != null
-                                            && !overrideConfiguration.messageSystemAttributeName().isEmpty()
-                                            ? overrideConfiguration.messageSystemAttributeName()
+        this.messageSystemAttributeNames = builder.messageSystemAttributeNames != null
+                                            ? builder.messageSystemAttributeNames
                                             : MESSAGE_SYSTEM_ATTRIBUTE_NAMES_DEFAULT;
 
-        this.receiveMessageAttributeNames = overrideConfiguration != null
-                                            && overrideConfiguration.receiveMessageAttributeNames() != null
-                                            && !overrideConfiguration.receiveMessageAttributeNames().isEmpty()
-                                            ? overrideConfiguration.receiveMessageAttributeNames()
+        this.receiveMessageAttributeNames = builder.receiveMessageAttributeNames != null
+                                            ? builder.receiveMessageAttributeNames
                                             : RECEIVE_MESSAGE_ATTRIBUTE_NAMES_DEFAULT;
 
-        this.adaptivePrefetching = overrideConfiguration != null && overrideConfiguration.adaptivePrefetching() != null
-                                   ? overrideConfiguration.adaptivePrefetching()
+        this.adaptivePrefetching = builder.adaptivePrefetching != null
```

Comment:
Since we removed `adaptivePrefetching` from the public config pojo, should we remove it from this builder as well?

Reply:
sure , removed adaptivePrefetching and its usage.
Had kept it just for testing old behavior that was in V1